pkg/proxy/iptables/proxier.go

```diff
@@ -538,6 +539,9 @@ func (proxier *Proxier) OnEndpointsAdd(endpoints *v1.Endpoints) {
 
 func (proxier *Proxier) OnEndpointsUpdate(oldEndpoints, endpoints *v1.Endpoints) {
 	if proxier.endpointsChanges.Update(oldEndpoints, endpoints) && proxier.isInitialized() {
+		if proxier.healthzServer != nil {
```

Comment:
could wrap proxyconfig.EndpointsHandler so each proxier didn't need this if we decided it was useful.